### PR TITLE
feat: add forget password feature with redux form

### PR DESCRIPTION
**Pull request by QaseemLodhi**

### CODE REVIEW

File: client/src/components/ForgotPassword/ForgotPasswordValidator.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { isEmail, isLowercase, isLength, isString } from 'validator';
```
Comment (mashhoodr) [inline client/src/components/ForgotPassword/ForgotPasswordValidator.js]: i cant see all of these being used

File: client/src/containers/ForgotPassword/ForgotPassword.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import './ForgotPassword.css';
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+
+class ForgotPassword extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+  static getDerivedStateFromProps(props) {
```
Comment (mashhoodr) [inline client/src/containers/ForgotPassword/ForgotPassword.js]: we should be adding typing now?

File: client/src/reducers/index.js [hunk 3]
```diff
@@ -60,6 +61,22 @@ const userState = (state = defaultUserState, action) => {
   }
 };
 
+const defaultResetPasswordState = {
+  message: null,
+};
+const resetPasswordState = (state = defaultResetPasswordState, action) => {
+  switch (action.type) {
+    case RESET_PASSWORD:
```
Comment (mashhoodr) [inline client/src/reducers/index.js]: Im not sure if we need to create reduce for forgot password?

File: client/src/containers/ForgotPassword/ForgotPassword.js (hunk 2 above)
Comment (QaseemLodhi) [reply client/src/containers/ForgotPassword/ForgotPassword.js]: Done

File: client/src/components/ForgotPassword/ForgotPasswordValidator.js (hunk 1 above)
Comment (QaseemLodhi) [reply client/src/components/ForgotPassword/ForgotPasswordValidator.js]: Done.

**Pull request merged**